### PR TITLE
Don't use a nonliteral format string in test/fheap.c:begin_test().

### PR DESCRIPTION
**Pull request by gnuoyd**

### CODE REVIEW

File: test/fheap.c [hunk 1]
```diff
@@ -547,28 +531,19 @@ get_fill_size(const fheap_test_param_t *tparam)
...
+    const char *del_str = get_del_string(tparam);
 
     /*
      * Test filling & removing all (small) objects from root direct block of absolute heap
      */
-    del_str = get_del_string(tparam);
-    HDassert(del_str);
-    test_desc = (char *)H5MM_malloc(HDstrlen(del_str) + HDstrlen(base_desc));
-    HDsprintf(test_desc, base_desc, del_str);
+    size_t test_desc_len = strlen(base_desc) + sizeof(" ") + strlen(del_str);
```
Comment (jhendersonHDF) [inline test/fheap.c:543]: Think this might need an extra +1 for the NUL terminator.

File: test/fheap.c [hunk 2]
```diff
@@ -466,40 +466,24 @@ add_obj(H5HF_t *fh, size_t obj_off, size_t obj_size, fheap_heap_state_t *state,
...
+static const char *
 get_del_string(const fheap_test_param_t *tparam)
 {
-    char *str;
-
     /* Remove half of total objects from heap */
     if (tparam->del_dir == FHEAP_DEL_FORWARD)
         if (tparam->drain_half == FHEAP_DEL_DRAIN_ALL)
-            str = H5MM_strdup("(all - forward)");
+            return "(all - forward)";
```
Comment (jhendersonHDF) [inline test/fheap.c:476]: I'm all for removing memory allocations where they aren't really needed!

**Comment by gnuoyd** [comment]
On Fri, Jan 28, 2022 at 02:47:17PM -0800, jhendersonHDF wrote:
> @jhendersonHDF commented on this pull request.
> 
> 
> 
> >  
>      /*
>       * Test filling & removing all (small) objects from root direct block of absolute heap
>       */
> -    del_str = get_del_string(tparam);
> -    HDassert(del_str);
> -    test_desc = (char *)H5MM_malloc(HDstrlen(del_str) + HDstrlen(base_desc));
> -    HDsprintf(test_desc, base_desc, del_str);
> +    size_t test_desc_len = strlen(base_desc) + sizeof(" ") + strlen(del_str);
> 
> Think this might need an extra +1 for the NUL terminator.

sizeof(" ") includes the NUL terminator of " ".

Dave

**Comment by jhendersonHDF** [comment]
> sizeof(" ") includes the NUL terminator of " ". Dave

Ah that's true. I mistakenly read as strlen(" ").

**Review by jhendersonHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**